scylla-server-rust/src/controller/data_controller.rs

```diff
@@ -0,0 +1,42 @@
+use axum::{
+    extract::{Path, State},
+    Json,
+};
+use serde::Serialize;
+
+use crate::{
+    error::ScyllaError,
+    prisma::{self},
+    services::data_service,
+    Database,
+};
+
+#[derive(Serialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct DataSend {
+    pub time: i64,
+    pub values: Vec<String>,
+}
+
+// convert the prisma type to the client type for JSON encoding
+impl From<&prisma::data::Data> for DataSend {
+    fn from(value: &prisma::data::Data) -> Self {
+        DataSend {
+            values: value.values.iter().map(|f| f.to_string()).collect(),
+            time: value.time.timestamp_millis(),
+        }
+    }
+}
```

Comment:
looks like this method is basically acting as the transformer? I would rephrase the comment to reflect that this is a method with that purpose, but IMO it's fine to keep these in the controller files

Reply:
I would disagree, i think they should be in their own file for nested cases like what if nodes contain data types which contain data, their purpose shouldnt be isolated to the controller